src/library/atoms/Input.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+
+import { InformationCircleIcon } from "@heroicons/react/24/outline";
+import { useState } from "react";
+import classNames from "helpers/classNames";
+
+type Props = {
+  type?: "text" | "email" | "password" | "number" | "search" | "tel" | "url";
+  label?: string;
+  id: string;
+  ariaDescribedBy?: string;
+  description?: string;
+  disabled?: boolean;
+  required?: boolean;
+  autoComplete?: string;
+};
+
+const isEmailValid = (email: string) => {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return emailRegex.test(email);
+};
+const isNotEmpty = (value: string) => value.trim() !== "";
+
+const Input = ({
+  type,
+  label = "Titulek",
+  id,
+  ariaDescribedBy,
+  description = "Popisek",
+  disabled = false,
+  required = false,
+  autoComplete,
+}: Props) => {
+  const [enteredValue, setEnteredValue] = useState<string>("");
+  const [isTouched, setIsTouched] = useState<boolean>(false);
+
+  const validateValue = (value: string) => {
+    if (type === "email") {
+      return isEmailValid(value);
+    }
+
+    return isNotEmpty(value);
+  };
+
+  const valueIsValid = validateValue(enteredValue);
+  const error = !valueIsValid && isTouched;
```

Comment:
Validace bych řešil až na úrovni formuláře, ne na úrovni fieldu. Používáme na to `zod`, viz login form. 

```
const formSchema = z.object({
  username: z.string().nonempty("E-mail je povinný").email(),
  password: z.string().nonempty("Heslo je povinné").min(6, "Heslo musí být dlouhé alespoň 6 znaků"),
});
```

Reply:
Stejně tak isTouched a enteredValue, onBlur a onFocus. To všechno za nás řeší react-hook-form